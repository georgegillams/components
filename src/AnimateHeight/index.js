import AnimateHeight from './AnimateHeight';

export default AnimateHeight;
export { AnimateHeight };
