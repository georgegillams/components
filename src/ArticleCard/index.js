import ArticleCard, { ARTICLE_CARD_LAYOUTS } from './ArticleCard';

export default ArticleCard;
export { ArticleCard, ARTICLE_CARD_LAYOUTS };
