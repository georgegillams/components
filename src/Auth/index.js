import AdminOnly from './AdminOnly';
import APIEntity from './APIEntity';
import ObjectAsList from './ObjectAsList';
import DebugObject from './DebugObject';
import EmailVerifiedOnly from './EmailVerifiedOnly';
import LoadingCover from './LoadingCover';
import LoggedInOnly from './LoggedInOnly';
import LoggedOutOnly from './LoggedOutOnly';
import Status from './Status';
import User from './User';

export default {
  APIEntity,
  AdminOnly,
  DebugObject,
  LoadingCover,
  EmailVerifiedOnly,
  LoggedInOnly,
  LoggedOutOnly,
  ObjectAsList,
  Status,
  User,
};
export {
  APIEntity,
  AdminOnly,
  DebugObject,
  LoadingCover,
  EmailVerifiedOnly,
  LoggedInOnly,
  LoggedOutOnly,
  ObjectAsList,
  Status,
  User,
};
