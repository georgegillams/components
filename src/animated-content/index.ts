export { default, ANIMATIONS } from './animated-content';
