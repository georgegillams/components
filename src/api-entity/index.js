export { default } from './api-entity';
