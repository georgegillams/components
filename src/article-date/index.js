export { default } from './article-date';
