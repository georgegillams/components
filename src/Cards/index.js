import Card from './Card';
import ArticleCard, { ARTICLE_CARD_LAYOUTS } from './ArticleCard';

export default { Card, ArticleCard, ARTICLE_CARD_LAYOUTS };
export { Card, ArticleCard, ARTICLE_CARD_LAYOUTS };
