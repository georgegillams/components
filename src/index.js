import {
  APIEntity,
  AdminOnly,
  DebugObject,
  LoadingCover,
  EmailVerifiedOnly,
  LoggedInOnly,
  LoggedOutOnly,
  ObjectAsList,
  Status,
  User,
} from './Auth';
import { Card, ArticleCard, ARTICLE_CARD_LAYOUTS } from './Cards';
import { Code, CodeInline, CodeBashArrow } from './Code';
import { CreativeCommons } from './CreativeCommons';
import { DegreeModule } from './Degree';
import { ColourPalleteItem } from './Design';
import { Footer } from './Footer';
import { Button, BurgerButton } from './Button';
import { LoadingIndicator } from './LoadingIndicator';
import { Logo } from './Logo';
import { MoneyPot } from './MoneyPot';
import { NavigationBar, NavigationItem } from './NavigationBar';
import {
  NotificationCollection,
  NOTIFICATION_TYPES,
  NotificationComp,
} from './Notifications';
import { Progress } from './Progress';
import { GGRedirect } from './Redirect';
import { RequestStatusContainer, RequestStatus } from './RequestStatus';
import {
  Skeleton,
  ButtonSkeleton,
  CardSkeleton,
  CheckboxSkeleton,
  CompactCardSkeleton,
  InputSkeleton,
  NotificationSkeleton,
  ProgressSkeleton,
  SectionSkeleton,
  SmallButtonSkeleton,
  SmallProgressSkeleton,
  SubSectionSkeleton,
  TextLinkSkeleton,
  TicketStatusSkeleton,
} from './Skeletons';
import { withScroll } from './ScrollContainer';
import { Tag, TagFilter, TAG_TYPES } from './Tag';
import {
  ArticleDate,
  Quote,
  Section,
  SubSection,
  MarkdownRenderer,
  AnimatedContent,
  TextLink,
} from './Typography';

export {
  APIEntity,
  ARTICLE_CARD_LAYOUTS,
  AdminOnly,
  AnimatedContent,
  ArticleCard,
  ArticleDate,
  BurgerButton,
  Button,
  ButtonSkeleton,
  Card,
  CardSkeleton,
  CheckboxSkeleton,
  Code,
  CodeBashArrow,
  CodeInline,
  ColourPalleteItem,
  CompactCardSkeleton,
  CreativeCommons,
  DebugObject,
  DegreeModule,
  EmailVerifiedOnly,
  Footer,
  GGRedirect,
  InputSkeleton,
  LoadingCover,
  LoadingIndicator,
  LoggedInOnly,
  LoggedOutOnly,
  Logo,
  MarkdownRenderer,
  MoneyPot,
  NOTIFICATION_TYPES,
  NavigationBar,
  NavigationItem,
  NotificationCollection,
  NotificationComp,
  NotificationSkeleton,
  ObjectAsList,
  Progress,
  ProgressSkeleton,
  Quote,
  RequestStatus,
  RequestStatusContainer,
  Section,
  SectionSkeleton,
  Skeleton,
  SmallButtonSkeleton,
  SmallProgressSkeleton,
  Status,
  SubSection,
  SubSectionSkeleton,
  TAG_TYPES,
  Tag,
  TagFilter,
  TextLink,
  TextLinkSkeleton,
  TicketStatusSkeleton,
  User,
  withScroll,
};
