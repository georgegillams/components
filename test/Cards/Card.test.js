import React from 'react';
import renderer from 'react-test-renderer';

import { Card } from '../../src/Cards';

describe('Card', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Card>Some content</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
