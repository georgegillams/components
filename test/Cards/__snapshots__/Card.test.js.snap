// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<button
  aria-disabled={null}
  className="sc-bczRLJ fxjpXu"
  disabled={false}
  onClick={null}
  onKeyPress={[Function]}
  type="button"
>
  <div
    className="sc-dkzDqf eUJUoo"
    disabled={false}
  >
    <div
      className="sc-hKMtZM kvlGVY"
    />
    <div
      className="sc-eCYdqJ heBWoi"
    >
      Some content
    </div>
  </div>
</button>
`;
